src/odemis/driver/tescan.py

```diff
@@ -1096,6 +1106,58 @@ def _pollXYZ(self):
         except Exception:
             logging.exception("Unexpected failure during XYZ polling")
 
+    def _checkPosition(self, ref_pos: Dict[str, float], pos: Dict[str, float], rel: bool = False):
+        """
+        Checks if the current stage position is equal to the given position.
+        :param ref_pos: stage position before movement
+        :param pos: requested stage position in relative or absolute coordinates
+        :param rel: By default False, requested stage position is in absolution position otherwise relative
+        """
+        # Get the target position in absolute coordinates
+        if rel:
+            target_pos = {ax: val + pos[ax] for ax, val in pos.items()}
+        else:
+            target_pos = {ax: pos[ax] for ax, val in pos.items()}
+
+        # Drop axes from the original position, which are not important because they have not moved
+        orig_pos = {a: ref_pos[a] for a, nv in target_pos.items() if nv != ref_pos[a]}
+
+        # TODO: base the timeout on stage movement time estimation, needs to be checked on hardware
+        wait_stage_move = 5  # s
```

Comment:
Let's make this an argument (with default value  = 5s). That should make it clearer that the function will time-out.